backend/src/repository/analysis_collection.py

```diff
@@ -310,16 +308,30 @@ def attach_third_party_link(self, analysis_name: str, third_party_enum: str, lin
         else:
             raise ValueError(f"Third party link type {third_party_enum} is not supported")
 
-        if third_party_enum not in analysis:
-            analysis[third_party_enum] = None
+        if 'third_party_links' not in analysis:
```

Comment:
Now that the code is getting refactored, the if/elif above to convert the text to an enumeration isn't necessary at https://github.com/uab-cgds-worthey/rosalution/pull/70/files#diff-87362c2167dc8433706de8b99deaf5b23d1da0c2b8ef5ab931bb42ec5bf72ab0R304.

An example of using Enums in path params is here at the following link. https://fastapi.tiangolo.com/sq/tutorial/path-params/#create-an-enum-class.

This will require an update to https://github.com/uab-cgds-worthey/rosalution/blob/main/backend/src/routers/analysis_router.py#L230 to cover the path param of the type from a string to the enumeration type.